sympy/core/function.py

```diff
@@ -863,6 +867,15 @@ class UndefinedFunction(FunctionClass):
     def __new__(mcl, name, bases=(AppliedUndef,), __dict__=None, **kwargs):
         __dict__ = __dict__ or {}
         # Allow Function('f', real=True)
+        # and/or Function(Symbol('f', real=True))
+        if hasattr(name, "assumptions0"):
+            for k, v in name.assumptions0.items():
+                if k in kwargs and v != kwargs[k]:
+                    raise ValueError(
+                        "clash between assumptions inherited from name and passed assumptions"
+                    )
+                else:
```

Comment:
I think we can remove this `else`. If  `if` doesn't work then `kwargs[k] = v` will. In other case code will stop execution due to `raise`.

Reply:
👍